src/app/course-page/directives/style.directive.ts

```diff
@@ -0,0 +1,37 @@
+import { Directive, ElementRef, HostBinding, HostListener, Renderer2, Input } from '@angular/core';
+import { CourseItem } from '../models/course-item.model';
+
+@Directive({
+  selector: '[app-course-item-style]'
+})
+export class StyleDirective {
+  @Input('courseItem') courseItem : CourseItem;
+
+  @HostBinding('class.highlighted') highlightedClass = null;
+
+  public currentDate = Date.now();
+  public date = new Date();
+  public freshCourseDate: number = this.date.setDate( this.date.getDate() - 14 );
+
+  constructor( private elRef: ElementRef, private r: Renderer2) {
```

Comment:
do not make short names, make understandable names. rename elRef to ElementRef

Reply:
Renamed elRef to ElementRef